aries/aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -428,38 +331,36 @@ impl BaseAnonCreds for IndyCredxAnonCreds {
         };
 
         let str_rev_reg_info = serde_json::to_string(&rev_reg_info)?;
-
-        wallet
-            .add_wallet_record(
-                CATEGORY_REV_REG_INFO,
-                &rev_reg_id.0,
-                &str_rev_reg_info,
-                None,
-            )
-            .await?;
+        let record = RecordBuilder::default()
```

Comment:
`Record` is using `derive_builder` - we would far more benefit here using `typed-builder` (used in the crate `messages`) as it gives you compile time guarantees you are using it right and it doesn't every return `Err` result. Obivously you don't get runtime flexibility, but I don't that's a property of the the `RecordBuilder` being leveraged

Switching from `derive_builder` to `typed-builder` should be possible with minimum changes, as they produce similar builder APIs.